src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -6324,71 +6331,218 @@ void CodeGen::genZeroInitFrame(int untrLclHi, int untrLclLo, regNumber initReg,
         noway_assert(uCntBytes == 0);
 
 #elif defined(TARGET_XARCH)
-        /*
-            Generate the following code:
-
-                lea     edi, [ebp/esp-OFFS]
-                mov     ecx, <size>
-                xor     eax, eax
-                rep     stosd
-         */
-
-        noway_assert(regSet.rsRegsModified(RBM_EDI));
-
-#ifdef UNIX_AMD64_ABI
-        // For register arguments we may have to save ECX and RDI on Amd64 System V OSes
-        if (intRegState.rsCalleeRegArgMaskLiveIn & RBM_RCX)
+        assert(compiler->getSIMDSupportLevel() >= SIMD_SSE2_Supported);
+        emitter*  emit        = GetEmitter();
+        regNumber frameReg    = genFramePointerReg();
+        regNumber zeroReg     = REG_NA;
+        int       blkSize     = untrLclHi - untrLclLo;
+        int       minSimdSize = XMM_REGSIZE_BYTES;
+
+        assert(blkSize >= 0);
+        noway_assert((blkSize % sizeof(int)) == 0);
+        assert((genRegMask(initReg) & intRegState.rsCalleeRegArgMaskLiveIn) == 0); // initReg is not a live
+// incoming argument reg
+#if defined(TARGET_64BIT) && defined(FEATURE_SIMD) && ALIGN_SIMD_TYPES
```

Comment:
None of this is tied to SIMD or SIMD alignment.

Using `TARGET_64BIT` here may lead to a bit of local confusion as we're already under `TARGET_XARCH`. 

```suggestion
#if defined(TARGET_ARM64)
```

(similar elsewhere)

Reply:
`(TARGET_AMD64)` rather than `(TARGET_ARM64)` I think?

Changed

Reply:
Hmm... seem to have broken something quite badly